src/wp-includes/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -0,0 +1,191 @@
+
+	/**
+	 * An array of CSS declarations (property => value pairs).
+	 *
+	 * @since 6.1.0
+	 *
+	 * @var array
+	 */
+	protected $declarations = array();
+
+	/**
+	 * Constructor for this object.
+	 *
+	 * If a `$declarations` array is passed, it will be used to populate
+	 * the initial $declarations prop of the object by calling add_declarations().
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $declarations An array of declarations (property => value pairs).
+	 */
+	public function __construct( $declarations = array() ) {
+		$this->add_declarations( $declarations );
+	}
+
+	/**
+	 * Add a single declaration.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $property The CSS property.
+	 * @param string $value    The CSS value.
+	 *
+	 * @return WP_Style_Engine_CSS_Declarations Returns the object to allow chaining methods.
+	 */
+	public function add_declaration( $property, $value ) {
+
+		// Sanitize the property.
+		$property = $this->sanitize_property( $property );
+		// Bail early if the property is empty.
+		if ( empty( $property ) ) {
+			return $this;
+		}
+
+		// Trim the value. If empty, bail early.
+		$value = trim( $value );
+		if ( '' === $value ) {
+			return $this;
+		}
+
+		// Add the declaration property/value pair.
+		$this->declarations[ $property ] = $value;
+
+		return $this;
+	}
+
+	/**
+	 * Remove a single declaration.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $property The CSS property.
+	 *
+	 * @return WP_Style_Engine_CSS_Declarations Returns the object to allow chaining methods.
+	 */
+	public function remove_declaration( $property ) {
+		unset( $this->declarations[ $property ] );
+		return $this;
+	}
+
+	/**
+	 * Add multiple declarations.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $declarations An array of declarations.
+	 *
+	 * @return WP_Style_Engine_CSS_Declarations Returns the object to allow chaining methods.
+	 */
+	public function add_declarations( $declarations ) {
+		foreach ( $declarations as $property => $value ) {
+			$this->add_declaration( $property, $value );
+		}
+		return $this;
+	}
+
+	/**
+	 * Remove multiple declarations.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $properties An array of properties.
+	 *
+	 * @return WP_Style_Engine_CSS_Declarations Returns the object to allow chaining methods.
+	 */
+	public function remove_declarations( $properties = array() ) {
+		foreach ( $properties as $property ) {
+			$this->remove_declaration( $property );
+		}
+		return $this;
+	}
+
+	/**
+	 * Get the declarations array.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return array
+	 */
+	public function get_declarations() {
+		return $this->declarations;
+	}
+
+	/**
+	 * Filters a CSS property + value pair.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $property The CSS property.
+	 * @param string $value    The value to be filtered.
+	 * @param string $spacer   The spacer between the colon and the value. Defaults to an empty string.
+	 *
+	 * @return string The filtered declaration as a single string.
```

Comment:
Should also mention when it return an empty string

Reply:
i.e. ", or an empty string."

Also, `as a single string` seems redundant as `$value` is documented as a string. Is there more to this that I'm missing?